librad/src/git/types.rs

```diff
@@ -114,6 +115,19 @@ impl<R> SymbolicRef<R> {
     }
 }
 
+/// If we want to be able to use a collection of [`Refspec`]s, where the remote
+/// portions don't line up, then we need to be able to talk about them using
+/// dynamically. In Rust this cab be done via `Box<dyn MyTrait>`.
+///
+/// In this case, we want to be able to turn [`Refspec`]s into `String`s. To
+/// avoid allowing anything that is `ToString` or `Display`, we opt for adding a
+/// `ToString`-like trait called `AsRefspec`.
+///
+/// **N.B.**: The only implementor of this trait should `Refspec`.
```

Comment:
Ok, well, I guess this is good for now. You can enforce that nothing else but `Refspec` can be `AsRefspec` by going through the gymnastics of defining a [sealed trait](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed)

Reply:
Nice trick :ok: :cartwheeling: